.dockerignore

```diff
@@ -1,4 +1,3 @@
-node_modules
```

Comment:
why don't we want to ignore the node_modules dir anymore?

Reply:
Because: 
1) Travis checks out the repo (no node_modules installed)
2) Travis installs/updates all npm dependencies (node_modules available + up to date)
3) We are ignoring node_modules for the docker build
4) We're installing/updating all npm deps as of the docker build

Last one can be skipped if node_modules are not ignored anymore and thus speed up the docker build step by ~1-2mins.

Reply:
To be more precise, docker build does not ignore the `node_modules` directory, but it removes it before build when not in travis context. This means that when you build the image outside travis, if you have any node modules installed in the directory, they will be copied over in the build context, and then removed. Maybe this could be improved by changing the line `ADD . ./nodeApp` to not copy the `node_modules` directory? Not sure if you can, or if it improves anything really, it's a small thing anyway.

Speeding up the build process by not ignoring the node_modules may be worthwhile, but I don't think it applies to the docker hub context. Also, ignoring/removing the node_modules will result to a more predictable image building, especially after we upgrade to npm 5 and start using locked package versions.

Reply:
I don't see why removing the node_modules folder will result in a more predictable image....At least for Travis, we're installing always the latest available version of a package. If you built it locally, it does the exact same.
Btw.: You can also restrict package versions now as of the package.json file, by not using the '^' operator or exchanging it with another operator (like `~`). The current default behaviour (`^X.Y.Z`) is to update to the latest minor version, but not update any major version (v major.minor.patch).

Removing node_modules folder or not including it at all will only result in 1-4 secs. I think it's not worth the effort.